Sources/web3swift/Tokens/ERC1400/Web3+ERC1400.swift

```diff
@@ -64,23 +64,14 @@ protocol IERC1400: IERC20 {
 // can be imperatively read and saved
 // FIXME: Rewrite this to CodableTransaction
 public class ERC1400: IERC1400, ERC20BaseProperties {
-
-    internal var _name: String?
-    internal var _symbol: String?
-    internal var _decimals: UInt8?
-    internal var _hasReadProperties: Bool = false
-
+    public private(set) var basePropertiesProvider: ERC20BasePropertiesProvider
     public var transaction: CodableTransaction
     public var web3: Web3
     public var provider: Web3Provider
     public var address: EthereumAddress
     public var abi: String
 
-    lazy var contract: Web3.Contract = {
-        let contract = self.web3.contract(self.abi, at: self.address, abiVersion: 2)
-        precondition(contract != nil)
-        return contract!
-    }()
+    public let contract: Web3.Contract
```

Comment:
@JeneaVranceanu you gave me this idea. This should work and not break anything because it was a computed var.

Also, much cleaner in my opinion was then already the `init` would fail - if something is wrongly configured.

Reply:
True, it should not break anything as it wasn't a public variable.
But computed variables still allow values being set to them later. It's the fact that it was just `lazy var` and not `public lazy var` that allows us to make it `public let` without breaking changes.

Anyway, this is good.